### PR TITLE
Order assessments by title if their dates are same

### LINKED ISSUE #1812: Further order assessments by title if their dates are same

### PR DESCRIPTION
**Pull request by allenwq**
Fix #1812 

Should consider using a new column to store the hash, instead of the milliseconds part of the start/end_at

### CODE REVIEW

File: spec/models/course/assessment_spec.rb [hunk 1]
```diff
@@ -269,15 +269,33 @@
...
     end
 
-    describe '.ordered_by_date' do
-      let(:other_assessment) { create(:assessment, *assessment_traits, course: course) }
+    describe '.ordered_by_date_and_title' do
+      let(:course) { create(:course) }
+      let(:start_at) { DateTime.new(2017, 1, 1).utc }
+      let!(:assessment1) do
+        assessment = create(:assessment, title: 'A', course: course)
+        # Use update_column to skip callbacks, this is to make sure the time in db is late than
```
Comment (fonglh) [inline spec/models/course/assessment_spec.rb]: "later than"

File: spec/models/course/assessment_spec.rb [hunk 2]
```diff
@@ -269,15 +269,33 @@
...
+        assessment.lesson_plan_item.update_column(:start_at, start_at + 1.second)
         assessment
-        other_assessment
-        consecutive = course.assessments.each_cons(2)
-        expect(consecutive.to_a).not_to be_empty
-        expect(consecutive.all? { |first, second| first.start_at <= second.start_at })
+      end
+
+      it 'orders the assessments by date and title' do
+        # Even assessment1's start_at is slightly late than assessment2, it should still come first.
```
Comment (fonglh) [inline spec/models/course/assessment_spec.rb]: "Even if assessment1's start_at is slightly later than assessment2, ..."

File: spec/models/course/assessment_spec.rb [hunk 3]
```diff
@@ -269,15 +269,33 @@
...
+        # This is to test the milliseconds part is ignored.
```
Comment (fonglh) [inline spec/models/course/assessment_spec.rb]: "This is to test that the milliseconds part of the time is ignored."

**Comment by allenwq** [comment]
Further updated

**Pull request merged**